library/ssl_tls.c

```diff
@@ -11271,22 +11267,40 @@ static int ssl_write_real( mbedtls_ssl_context *ssl,
          * copy the data into the internal buffers and setup the data structure
          * to keep track of partial writes
          */
-        ssl->out_msglen  = len;
+        ssl->out_msglen = len;
         ssl->out_msgtype = MBEDTLS_SSL_MSG_APPLICATION_DATA;
-        mbedtls_platform_memcpy( ssl->out_msg, buf, len );
+        mbedtls_platform_memcpy(ssl->out_msg, buf, len);
 
-        if( ( ret = mbedtls_ssl_write_record( ssl, SSL_FORCE_FLUSH ) ) != 0 )
+#if defined(MBEDTLS_FI_COUNTERMEASURES) && !defined(MBEDTLS_SSL_CBC_RECORD_SPLITTING)
+        /* Secure against buffer substitution */
+        if (buf == ssl->out_msg_dup &&
```

Comment:
This check compares pointer to `buf` with a pointer to `ssl->out_msg_dup`. How is this supposed to work?

Reply:
@AndrzejKurek , this is the same user buf pointer that was stored in context and i want to be sure that duplicated pointers are still the same

Reply:
```suggestion
        if( buf == ssl->out_msg_dup &&
```